### PR TITLE
Update VMC status conditions

### PR DESCRIPTION
**Pull request by desagar**
# Description

Add conditions to the VMC status and update condition type Ready = true when the VMC is ready to be used, so that users can wait on the condition.

Fixes VZ-2369

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/clusters/vmc_controller.go [hunk 1]
```diff
@@ -250,6 +254,51 @@ func (r *VerrazzanoManagedClusterReconciler) reconcileManagedClusterDelete(ctx c
...
+			matchingCondition = &existingCondition
+		}
+	}
+	if matchingCondition == nil {
+		vmc.Status.Conditions = append(vmc.Status.Conditions, condition)
+	} else {
+		matchingCondition.Message = condition.Message
+		matchingCondition.Status = condition.Status
+		matchingCondition.LastTransitionTime = condition.LastTransitionTime
+		r.log.Infof("VMC %s has existing condition with type %s = %s", vmc.Name, matchingCondition.Type, matchingCondition.Status)
```
Comment (mgianatagh) [inline platform-operator/controllers/clusters/vmc_controller.go:296]: should this be debugf?

File: platform-operator/controllers/clusters/vmc_controller.go [hunk 2]
```diff
@@ -250,6 +254,51 @@ func (r *VerrazzanoManagedClusterReconciler) reconcileManagedClusterDelete(ctx c
...
+	}
+	r.log.Infof("Updating Status of VMC %s with condition type %s = %s: %v", vmc.Name, condition.Type, condition.Status, vmc.Status.Conditions)
```
Comment (mgianatagh) [inline platform-operator/controllers/clusters/vmc_controller.go:298]: should this be debugf?

File: platform-operator/controllers/clusters/vmc_controller.go (hunk 1 above)
Comment (desagar) [reply platform-operator/controllers/clusters/vmc_controller.go]: changed to debugf

File: platform-operator/controllers/clusters/vmc_controller.go (hunk 2 above)
Comment (desagar) [reply platform-operator/controllers/clusters/vmc_controller.go]: yes, changed

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**